lib/pure/unicode.nim

```diff
@@ -1489,6 +1502,130 @@ proc capitalize*(s: string): string {.noSideEffect, procvar,
 
   result = $toUpper(rune) & substr(s, i)
 
+proc normalizeUnicode*(s: string): string {.noSideEffect,
```

Comment:
I think this should be called ``normalize``. Why is this procedure special? you don't add a ``unicode`` suffix to ``strip`` below or any other procs.

Reply:
This is special, because this proc introduces behaviour that is different to `normalize` in `strutils`. The `strutils` variant only does ASCII lowercasing by converting characters `in {'A'-'Z'}` to lowercase. Therefore the `strutils` variant is quicker (it has less to do).

The unicode implementation does proper lowercasing using the unicode `toLower` proc for each Rune in the string.

Other procs in `unicode` hav the same names in cases where the implementation is not really different to the `strutils` variant other than supporting `openarray[Rune]` as input and being unicode aware.

Reply:
Okay, so this is a bit peculiar.

The ``strutils.toLower`` proc was renamed to ``toLowerAscii``. Shouldn't we rename ``strutils.normalize`` to ``normalizeAscii`` and name this proc ``normalize``?

Reply:
Ah, yes normalize is special, since its useless is e.g. with nim identifiers. Therefore the more common use is the ASCII variant.

In the other cases, the Unicode variant is the better choice since it does a more thorough job than the ASCII version

Reply:
Alright. That's fine I suppose. But please explain this in the documentation, because someone will undoubtedly notice the inconsistency, complain about it and I won't remember the reason for it :)